### ISSUE TITLE
A couple of issues when using powerlevel9k on Android

### ISSUE THREAD

**Issue by guixxx** [bhilburn/powerlevel9k#479]
I have two issues using powerlevel9k on Termux. The first one is when when I first open a terminal window:

![screenshot_20170412-100930](https://cloud.githubusercontent.com/assets/27077338/24959185/505bc820-1f68-11e7-963d-6a893e4e29a3.png)

`powerlevel9k_prepare_prompts:3 number truncated after 10 digits: 99999999999`

I don't know if it affects anything, but that warning message is kinda annoying, tbh. The second issue is related to the `command_execution_time` plug-in. As you can see in the screenshot above, the plug-in is reporting the startup time wrongly, but after that, any command I run the plug-in reports the execution time correctly:

![screenshot_20170412-100949](https://cloud.githubusercontent.com/assets/27077338/24959188/541f56fc-1f68-11e7-9788-1be1665a22e2.png)

My `.zshrc`:

```zsh
DEFAULT_USER="${USER}"
DISABLE_AUTO_UPDATE=true
POWERLEVEL9K_INSTALLATION_PATH="${HOME}/.antigen/bundles/bhilburn/powerlevel9k"
POWERLEVEL9K_COMMAND_EXECUTION_TIME_THRESHOLD=0
POWERLEVEL9K_COMMAND_EXECUTION_TIME_BACKGROUND=245
POWERLEVEL9K_COMMAND_EXECUTION_TIME_FOREGROUND='black'
POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(status context dir vcs)
POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=(root_indicator background_jobs command_execution_time time)
POWERLEVEL9K_STATUS_VERBOSE=false
POWERLEVEL9K_MODE='nerdfont-fontconfig'
source $HOME/.antigen-git/antigen.zsh
antigen use oh-my-zsh
antigen bundle zsh-users/zsh-syntax-highlighting
antigen theme bhilburn/powerlevel9k powerlevel9k
antigen apply
```

I'm running Android 6.0.1 and using Termux as my emulator. Additional info:

```
$ zsh --version
zsh 5.3 (arm-unknown-linux-androideabi)
$ uname -a
Linux localhost 3.4.42-ge642ba1 #1 SMP PREEMPT Tue Dec 20 13:28:16 PST 2016 armv7l Android
```

Thank you.

Edit: fixed typos in .zshrc

**Comment by dritter** [comment]
😲  Wow. You are running ZSH on your phone? Impressive. Never thought about it.

Regarding the issues:
`powerlevel9k_prepare_prompts:3` is the following line:
```zsh
_P9K_COMMAND_DURATION=$((EPOCHREALTIME - _P9K_TIMER_START))
```
It calculates the duration of the current command. The `_P9K_TIMER_START` variable holds an initially big number that gets subtracted from `EPOCHREALTIME` (the current time). The segment gets only shown if the execution time was above a certain threshold. We do this, to avoid the segment from getting wrongly rendered initially.

Long story short: I'll install P9K on my phone, and fix the problems. :)

**Comment by guixxx** [comment]
I am! Termux is amazing, you should definitely give it a try. After a bit of digging I realized what was causing those warning messages.

Thanks

**Comment by onaforeignshore** [comment]
solution in PR #480

**Comment by bhilburn** [comment]
Closing this to continue the conversation in #480, per @onaforeignshore's comment above.
